libraries/wutsocket/addrinfo.c

```diff
@@ -0,0 +1,53 @@
+#include "wut_socket.h"
+#include <nsysnet/_netdb.h>
+#include <netdb.h>
+
+int
+getaddrinfo(const char *node,
+            const char *service,
+            const struct addrinfo *hints,
+            struct addrinfo **res)
+{
+   int rc;
+
+   if (!node && !service) {
+      return EAI_NONAME;
+   }
+
+   if (!res) {
+      errno = EINVAL;
+      return EAI_SYSTEM;
+   }
+
+   rc = RPLWRAP(getaddrinfo)(node, service, hints, res);
```

Comment:
Missing errno translation (`__wut_get_nsysnet_result`?) when the return value is `EAI_SYSTEM`.

Reply:
From looking at the nsysnet implementation, getaddrinfo never returns `EAI_SYSTEM`.
Errno translation is probably not necessary here.